src/plots/barPlot.ts

```diff
@@ -441,9 +440,31 @@ export module Plots {
             x += offset;
           }
 
+          var showLabel = true;
+          var labelPosition = {
+            x: x,
+            y: positive ? y : y + h - measurement.height
+          };
+
+          if (this._isVertical) {
+            labelPosition.x = x + w / 2 - measurement.width / 2;
+          } else {
+            if (!positive) {
+              labelPosition.x = x + w - measurement.width;
+            } else {
+              labelPosition.x = x;
+            }
+          }
+
+          if (labelPosition.x < 0 || labelPosition.x + measurement.width > +this._boundingBox.attr("width") ||
+              labelPosition.y < 0 || labelPosition.y + measurement.height > +this._boundingBox.attr("height")) {
```

Comment:
Instead of `this._boundingBox.attr("width")` and `this._boundingBox.attr("height")` you should be able to use `this.width()` and `this.height()`.